### PR TITLE
Telemetry: Adding in tests for Java Agent 1.29

### LINKED ISSUE #26919: Update tests for newer version of Telemetry Agent
Currently we're testing Telemetry Agent 1.19 with all versions of the mpTelemetry feature in `AgentTest`.

However, this means:
* We're not testing with the agent version which matches mpTelemetry-1.1
* None of the agent tests run on Java 21 which means they don't run in the personal or continuous builds so we don't find out if anything is broken until it hits the release branch and runs in the SOE or a build which uses a specific Java version.

### PR DESCRIPTION
**Pull request by jakub-pomykala**
- Telemetry with Java Agent 1.29 for MicroProfile 5 and MicroProfile 6.1.
- MicroProfile 6.0 uses Java Agent 1.19
- MicroProfile 1.4 and 4.1 are currently not working with Java Agent 1.29 due to a bug in the Java Agent

This PR does not fix #26919 but is a half way point.

See https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/10159

### CODE REVIEW

File: dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/tests/Agent129Test.java [hunk 1]
```diff
@@ -0,0 +1,446 @@
+/*******************************************************************************
+ * Copyright (c) 2022,2023 IBM Corporation and others.
```
Comment (Yas-Franco) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/tests/Agent129Test.java]: Copyright headers need updating to 2024

File: dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/tests/AgentConfigMultiAppTest.java
Comment (Yas-Franco) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/tests/AgentConfigMultiAppTest.java:1]: Copyright header needs updating

File: dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/tests/AgentConfigTest.java [hunk 2]
```diff
@@ -183,14 +216,27 @@ public void testDisableInstrumentation() throws Exception {
...
+        }
 
         // We should only have one span, it should be the server span
         // No span was created for the WithSpan annotation
         assertThat(span, hasKind(SERVER));
     }
 
     @Test
+    /*
+     * Skipping for 1.4 and 4.1 as JavaAgent 1.29 currently will not return a span for methods annotated with @withSpan (BUG)
```
Comment (Yas-Franco) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/tests/AgentConfigTest.java]: Link issue in this comment

File: dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/tests/AgentTest.java
Comment (Yas-Franco) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/tests/AgentTest.java:1]: Update comment to say you are testing the java agent 1.19 only

**Review by Yas-Franco — COMMENTED** [review]
Minor changes needed but otherwise it looks good!

**Comment by jakub-pomykala** [comment]
#build #spawn.fullfat.buckets=io.openliberty.microprofile.telemetry.1.0.internal_fat,io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat,io.openliberty.microprofile.telemetry.1.0.internal_fat_tck,io.openliberty.microprofile.telemetry.1.1.internal_fat_tck

**Review by Yas-Franco — APPROVED** [review]

**Comment by Yas-Franco** [comment]
#libby

**Pull request merged**